deck/packages/core/src/pipeline/executions/executionGroup/ExecutionGroup.tsx

```diff
@@ -153,7 +154,9 @@ export class ExecutionGroup extends React.PureComponent<IExecutionGroupProps, IE
             .then((plan) => resolve(plan))
             .catch(() => resolve(config));
         } else {
-          resolve(config);
+          PipelineConfigService.getPipelineForApplication(this.props.application.name, config.name)
+            .then((pipeline) => resolve(pipeline))
+            .catch(() => resolve(config));
```

Comment:
Can you explain this fallback case a bit more?  As in, what's the difference between `resolve(pipeline)` and `resolve(config)`?  Is it really OK to do a "best effort" thing here?

Reply:
Yeah, `config` here would be the potentially old pipeline. I figure if this fails, we just revert to the current behaviour (which most of the time works, unless the expected artifact changed between last load and now). If we make this throw an error it would be a breaking change, which feels a bit too strict. What do you think?